src/field_theory/algebraic_closure.lean

```diff
@@ -0,0 +1,283 @@
+/-
+Copyright (c) 2020 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import algebra.direct_limit
+import field_theory.splitting_field
+import analysis.complex.polynomial
+
+/-!
+# Algebraic Closure
+
+In this file we define the typeclass for algebraically closed fields and algebraic closures.
+We also construct an algebraic closure for any field.
+
+## Main Definitions
+
+- `is_alg_closed k` is the typeclass saying `k` is an algebraically closed field, i.e. every
+polynomial in `k` splits.
+
+- `is_alg_closure k K` is the typeclass saying `K` is an algebraic closure of `k`.
+
+- `algebraic_closure k` is an algebraic closure of `k` (in the same universe).
+
```

Comment:
Could you add a description of the construction you are using, and a bibliography reference if you followed a book or paper? There are many ways to prove this theorem and it would help to know which one you use.

Reply:
Done